auth-web/tests/unit/components/EntityManagement.spec.ts

```diff
@@ -173,4 +174,48 @@ describe('Entity Management Component', () => {
     expect(wrapper.find('#incorporate-numbered-unlimited-btn').exists()).toBe(true)
     expect(wrapper.find('#incorporate-numbered-ccc-btn').exists()).toBe(true)
   })
+
+  it('calls the nr success modal', async () => {
+    const mockedSyncBusinesses = jest.fn()
+    wrapper.vm.syncBusinesses = mockedSyncBusinesses
+    const mockedSearchNRIndex = jest.fn().mockReturnValue(0)
+    wrapper.vm.searchNRIndex = mockedSearchNRIndex
+    wrapper.vm.showAddSuccessModalNR('NR 1111111')
+    await flushPromises()
+
+    expect(mockedSyncBusinesses).toHaveBeenCalled()
+    expect(mockedSearchNRIndex).toHaveBeenCalled()
+    expect(wrapper.vm.snackbarText).toBe('NR 1111111 was sucessfully added to your table.')
+  })
+
+  it('calls the nr error modal', async () => {
+    const mockedNrErrorMethod = jest.fn()
+    wrapper.vm.$refs.errorDialog.open = mockedNrErrorMethod
+    wrapper.vm.showNRErrorModal()
+    expect(wrapper.vm.dialogTitle).toBe('Error Adding Name Request')
+    expect(wrapper.vm.dialogText).toBe(
+      'We couldn\'t find a name request associated with the phone number or email address you entered. Please try again.'
+    )
+    expect(mockedNrErrorMethod).toHaveBeenCalled()
+  })
+
+  it('renders snackbar visible 1 second after toggled', async () => {
+    wrapper.vm.showNRErrorModal()
+    await flushPromises()
+
+    setTimeout(async () => {
+      expect(wrapper.find('#success-nr-snackbar').exist()).toBe(true)
+      expect(wrapper.find('#success-nr-snackbar').isVisible()).toBe(true)
+    }, 1000)
+  })
+
+  it('renders snackbar invisible 5 seconds after toggled', async () => {
+    wrapper.vm.showNRErrorModal()
+    await flushPromises()
+
+    setTimeout(async () => {
+      expect(wrapper.find('#success-nr-snackbar').exist()).toBe(true)
+      expect(wrapper.find('#success-nr-snackbar').isVisible()).toBe(false)
+    }, 5000)
```

Comment:
This doesn't work, as the setTimeout function is called 5 seconds after this test ends.

One alternative is to call a "done" function in your setTimeout function.

Another way is to "await sleep(5000)" before your expect statements. (I don't know if sleep() or Sleep() is declared in this repo. If not, just add it to utils/.)

Reply:
This is fixed. Resolving.